superset/assets/src/dashboard/components/FilterIndicatorsContainer.jsx

```diff
@@ -78,29 +73,33 @@ export default class FilterIndicatorsContainer extends React.PureComponent {
           chartId,
           componentId,
           directPathToFilter,
-          scope,
           isDateFilter,
           isInstantFilter,
           columns,
           labels,
+          scopes,
         } = dashboardFilter;
 
-        // do not apply filter on filter_box itself
-        // do not apply filter on filterImmuneSlices list
-        if (
-          currentChartId !== chartId &&
-          !filterImmuneSlices.includes(currentChartId)
-        ) {
+        if (currentChartId !== chartId) {
           Object.keys(columns).forEach(name => {
```

Comment:
Can use `.filter` instead of adding condition to `.forEach`

```js
Object.keys(columns)
  .filter(name => getChartIdsInFilterScope({ filterScope: scopes[name] }).includes(currentChartId))
  .forEach(name => { ... })
```